src/ripple/core/SQLInterface_shards.h

```diff
@@ -0,0 +1,204 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2020 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CORE_SQLINTERGACE_SHARDS_H_INCLUDED
+#define RIPPLE_CORE_SQLINTERGACE_SHARDS_H_INCLUDED
+
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/app/misc/Manifest.h>
+#include <ripple/core/Config.h>
+#include <ripple/core/DatabaseCon.h>
+#include <ripple/core/SQLInterface.h>
+#include <ripple/overlay/PeerReservationTable.h>
+#include <ripple/peerfinder/impl/Store.h>
+#include <boost/filesystem.hpp>
+
+namespace ripple {
+
```

Comment:
These free function don't appear to be shard specific - they are called from the sqlite functions, for example. I'd move the non-shard specific functions to a file with a more generic name.

Reply:
All these function are potentially shard-specific, because they will be called for shard dbs if node db does not exist.

Reply:
If these functions are only used for shards, then it's OK to keep them in a file called `XXX_shards`. If they are used for shards and other uses I'd like to move those functions to a different file. For example, `getMinLedgerSeq` appears to used for more than just shards. If that's true (and it's possible I misread the code), then I'd propose moving all the functions that are used for more than just shards outside of this file.

Reply:
Done. File was splitted into 2 ones:*_shards and *_nodes